### PR TITLE
Add package of the week video

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@natebosch

**Comment by natebosch** [comment]
The video doesn't give much information. The video seems like a really good way to help more folks discover the package. After they know it exists and are looking at the docs I don't know what value they would get from a link to the video.

**Comment by tvolkert** [comment]
The package is listed in api.flutter.dev, so they may just click on it to see what it has to offer without any context.  For those users, I think the video still provides value.

**Comment by natebosch** [comment]
cc @lrhn WDYT?

**Comment by tvolkert** [comment]
fwiw, here's what users see right now:

async

File: lib/async.dart [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+/// {@youtube 560 315 https://www.youtube.com/watch?v=r0tHiCjW2w0}
+
```
Comment (lrhn) [inline lib/async.dart:6]: This is a doc comment, so you should probably add a `library` declaration to make it document the library.

We should also add a real library documentation, but for now, let's just have a headline before the link:
```dart
/// Various functionality associated with the asynchronous features of the `dart:async` library.
///
/// {@youtube 560 315 https://www.youtube.com/watch?v=r0tHiCjW2w0}
library pkg.async;
```

**Review by lrhn — APPROVED** [review]

File: lib/async.dart (hunk 1 above)
Comment (natebosch) [reply lib/async.dart:6]: > you should probably add a `library` declaration to make it document the library.

Dartdoc should handle it either way - but I'm in favor of the explicit library declaration for now.

I _really_ hope we can make that `library;` soon.

If I can nitpick the wording, I think "Various" and "functionality" aren't conveying much - virtually any package can be describe as having "Various functionality". How about:

```dart
/// Utilities that expand on the asynchronous features of the `dart:async` library.
///
/// {@youtube 560 315 https://www.youtube.com/watch?v=r0tHiCjW2w0}
library pkg.async;
```

**Review by natebosch — APPROVED** [review]
Adding the doc and link LGTM once it has a header in addition to the link.

File: lib/async.dart (hunk 1 above)
Comment (lrhn) [reply lib/async.dart:6]: That would LGTM

**Comment by tvolkert** [comment]
Ok, all done.  @natebosch can you merge for me?

**Pull request merged**